docs/source/howto/interact.rst

```diff
@@ -0,0 +1,112 @@
+.. _how-to:interact:
+
+**************************
+How to interact with AiiDA
+**************************
+
+There are a variety of manners to interact with AiiDA:
+
+ * :ref:`Through the command line interface <how-to:interact-cli>`
+ * :ref:`Through interactive shells <how-to:interact-shell>`
+ * :ref:`Through interactive notebooks <how-to:interact-notebook>`
+ * :ref:`Through scripts <how-to:interact-scripts>`
+ * :ref:`Through the REST API <how-to:interact-restapi>`
+
+
+.. _how-to:interact-cli:
+
+Command line interface
+======================
+
+AiiDA comes with a command line interface called ``verdi``.
+The :ref:`reference:command-line` section gives an overview of all available commands.
+For more detailed information, refer to the topic section :ref:`topics:cli`.
+
+
+.. _how-to:interact-shell:
+
+Interactive shells
+==================
+
+AiiDA provides a Python API that can be used from an interactive shell, such as `IPython <https://ipython.org/>`_.
+To use AiiDA from an IPython shell, one just has to load the profile as follows:
+
+.. code-block:: ipython
+
+    In [1]: from aiida import load_profile
+
+    In [2]: load_profile()
+    Out[2]: <aiida.manage.configuration.profile.Profile at 0x7fccfd6c50a0>
+
+One can pass a particular profile name to :meth:`~aiida.load_profile`, otherwise the default profile is loaded.
+Alternatively, the command line interface ``verdi`` provides a handy shortcut:
+
+.. code-block:: console
+
+    $ verdi shell
+
+This command will open a normal IPython shell but automatically loads the AiiDA profile.
+In addition, certain modules from AiiDA's API that are used very often are automatically imported.
+
+
+.. _how-to:interact-notebook:
+
+Interactive notebooks
+=====================
+
+Similar to :ref:`interactive shells <how-to:interact-shell>`, AiiDA is also directly compatbile with interactive Python notebooks, such as `Jupyter <https://jupyter.org/>`_.
+To use AiiDA's Python API in a notebook, first a profile has to be loaded:
+
+.. code-block:: ipython
+
+    In [1]: from aiida import load_profile
+
+    In [2]: load_profile()
+    Out[2]: <aiida.manage.configuration.profile.Profile at 0x7fccfd6c50a0>
+
+One can pass a particular profile name to :meth:`~aiida.load_profile`, otherwise the default profile is loaded.
+
+
+.. _how-to:interact-scripts:
+
+Scripts
+=======
+
+AiiDA's Python API can be used in Python scripts mixed with any other Python code.
+The only requirement, is that before the API is used, an AiiDA profile is loaded.
+This can be accomplished by adding the following lines to the script:
+
+.. code-block:: python
```

Comment:
This is a bit of repetition.
Can we start with this section, and then later when we discuss `ipython` and `jupyter` we just say that we need to use the same `load_profile` command already explained?

Reply:
How do you want to reorganize it though? I wanted the separate sections to be independent and not refer to one another to be fully understandable. I thought the repetition was worth it. How would you restructure it?